### PR TITLE
Fix crash in sampling terrain for points with undefined position

### PR DESCRIPTION
**Pull request by sacredbanana**
When sampling the terrain, if any points get an undefined position due to being outside of the rectangle, it would crash. Now just skip that point and continue sampling.

This would happen for sampling terrain in Antarctica where there was a hole in the terrain data if any points resided in that hole.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @sacredbanana! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/CesiumGS/cesium/blob/main/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/sacredbanana/cesium/blob/sample-terrain-crash-fix/CONTRIBUTORS.md) file!
* :x: Missing CLA.
    * Please send in a [Contributor License Agreement](https://github.com/CesiumGS/cesium/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) (CLA) and comment back here to let us know to check this!
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/sample-terrain-crash-fix/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by sacredbanana** [comment]
I signed the agreement

**Comment by sacredbanana** [comment]
@ggetz thanks! Committed your suggestion

**Comment by ggetz** [comment]
Great, thanks @sacredbanana! Can you add an item to [CHANGES.md](https://github.com/CesiumGS/cesium/blob/sample-terrain-crash-fix/CHANGES.md) describing the fix?

File: packages/engine/Source/Core/sampleTerrain.js [hunk 1]
```diff
@@ -132,6 +132,10 @@ function doSampling(terrainProvider, level, positions) {
   const tileRequestSet = {}; // A unique set
   for (i = 0; i < positions.length; ++i) {
     const xy = tilingScheme.positionToTileXY(positions[i], level);
+    if (!defined(xy)) {
```
Comment (ggetz) [inline packages/engine/Source/Core/sampleTerrain.js:135]: In order to make sure the code still builds, make sure to import `defined` at the top of this file.

Comment (sacredbanana) [reply packages/engine/Source/Core/sampleTerrain.js:136]: done thanks

**Comment by sacredbanana** [comment]
> Great, thanks @sacredbanana! Can you add an item to [CHANGES.md](https://github.com/CesiumGS/cesium/blob/sample-terrain-crash-fix/CHANGES.md?rgh-link-date=2023-01-03T14%3A25%3A51Z) describing the fix?

Added now :)

**Comment by ggetz** [comment]
Thanks @sacredbanana!

**Pull request merged**